stellargraph/core/graph.py

```diff
@@ -41,166 +41,232 @@
-    This allows node features to have different sizes for each node type::
+    If this data represents an undirected graph (the ordering of each edge source/target doesn't
+    matter)::
+
+        Gs = StellarGraph(nodes, edges)
 
-        node_data = {
-            node_type_1: pd.DataFrame(...),
-            node_type_2: pd.DataFrame(...),
-        }
-        Gs = StellarGraph(nx_graph, node_features=node_data)
 
+    If this data represents a directed graph (the ordering does matter)::
 
-    You can also supply the node feature vectors as an iterator of `node_id`
-    and feature vector pairs, for graphs with single and multiple node types::
+        Gs = StellarDiGraph(nodes, edges)
 
-        node_data = zip([node_id_1, node_id_2, ...],
-            [feature_vector_1, feature_vector_2, ..])
-        Gs = StellarGraph(nx_graph, node_features=node_data)
 
-    .. warning::
+    Numeric node features are taken as any columns of the nodes DataFrame. For example, if the graph
+    above has two features ``x`` and ``y`` associated with each node::
+
+        nodes = pd.DataFrame(
+            {"x": [-1, 2, -3, 4], "y": [0.4, 0.1, 0.9, 0]}, index=["a", "b", "c", "d"]
+        )
+
+    Edge weights are taken as the optional ``weight`` column of the edges DataFrame::
+
+        edges = pd.DataFrame({
+            "source": ["a", "b", "c", "d", "a"],
+            "target": ["b", "c", "d", "a", "c"],
+            "weight": [10, 0.5, 1, 3, 13]
+        })
+
+    Heterogeneous graphs, with multiple node or edge types, can be created by passing multiple
+    DataFrames in a dictionary. The dictionary keys are the names/identifiers for the type. For
+    example, if the graph above has node ``a`` of type ``foo``, and the rest as type ``bar``, the
+    construction might look like::
+
+        foo_nodes = pd.DataFrame({"x": [-1]}, index=["a"])
+        bar_nodes = pd.DataFrame(
+            {"y": [0.4, 0.1, 0.9], "z": [100, 200, 300]}, index=["b", "c", "d"]
+        )
+
+        StellarGraph({"foo": foo_nodes, "bar": bar_nodes}, edges)
 
-        The constructor variant using the ``nodes=..., edges=...`` arguments to create a "new"
-        StellarGraph is experimental: the type is insufficiently documented and does not support
-        some algorithms.
+    Notice the ``foo`` node has one feature ``x``, while the ``bar`` nodes have 2 features ``y`` and
+    ``z``. A heterogeneous graph can have different features for each type.
 
+    Edges of different types work in the same way. example instance, if edges have different types based
+    on their orientation::
+
+        horizontal_edges = pd.DataFrame(
+            {"source": ["a", "c"], "target": ["b", "d"]}, index=[0, 2]
+        )
+        vertical_edges = pd.DataFrame(
+            {"source": ["b", "d"], "target": ["c", "a"]}, index=[1, 3]
+        )
+        diagonal_edges = pd.DataFrame({"source": ["a"], "target": ["c"]}, index=[4])
```

Comment:
Is it important that there's no collision between edge indices across different edge types? I was under the impression that it wasn't, but this example kind of suggests that it is

Reply:
Ah I see - is it worth an explicit mention here then? Or you intend for this to be not the case soon?

Reply:
I'll add a note about it. I don't personally have any intentions for relaxing it, but I guess it might be a good idea 🤔